tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -1237,12 +1241,17 @@ SpirvInstruction *SpirvEmitter::doExpr(const Expr *expr,
   } else if (isa<CXXThisExpr>(expr)) {
     assert(curThis);
     result = curThis;
-  } else if (isa<CXXConstructExpr>(expr)) {
+  } else if (const auto *constructExpr = dyn_cast<CXXConstructExpr>(expr)) {
     // For RayQuery type, we should not explicitly initialize it using
     // CXXConstructExpr e.g., RayQuery<0> r = RayQuery<0>() is the same as we do
     // not have a variable initialization. Setting nullptr for the SPIR-V
     // instruction used for expr will let us skip the variable initialization.
-    if (!hlsl::IsHLSLRayQueryType(expr->getType()))
+    if (hlsl::IsVKBufferPointerType(expr->getType())) {
+      auto *arg = constructExpr->getArg(0);
+      result = spvBuilder.createConvertUToPtr(doExpr(arg), expr->getType());
+      result->setRValue(); // ???
```

Comment:
Yes, I believe you want to make this an rvalue. The constructor should not return a reference.